x/ibc-transfer/types/coin.go

```diff
@@ -41,8 +41,13 @@ func GetPrefixedDenom(portID, channelID, baseDenom string) string {
 }
 
 // GetPrefixedCoin creates a copy of the given coin with the prefixed denom
+// NOTE: this function is unsafe to use because it does not do sdk.Coin
```

Comment:
Maybe we rename or make it do the validation?

Reply:
Not really a big deal tho.

Reply:
I added this early into refactoring tests. I don't think it makes sense to even use this function since coins with full prefixed denoms should never exist (they don't ever exist internally in the SDK), I will remove it entirely